server.js

```diff
@@ -0,0 +1,155 @@
+'use strict';
+
+const bodyParser = require('body-parser');
+const mongoose = require('mongoose');
+var express = require('express');
+var app = express();
+require('isomorphic-fetch');
+
+const {DATABASE_URL, PORT} = require('./config');
+const {Show} = require('./models');
+
+mongoose.Promise = global.Promise;
+
+app.use(express.static('public'));
+app.use(bodyParser.json());
+
+
+app.get('/shows', (req, res) => {
+    Show
+    .find()
+    .exec()
+    .then(shows => {
+        res.json(shows.map(Show => Show.apiRepr()));
+    })
+     .catch(err => {
+         console.error(err);
+         res.status(500).json({error: 'something went terribly wrong'});
+     });
+
+});
+
+
+// When calling the POST endpoint, the inputted title is sent to theMovieDb API, then the response is saved in our database, along
+// with the inputted return date.
+
+app.post('/shows', (req, res) => {
+
+    const requiredFields = ['title', 'returns'];
+    for (let i=0; i<requiredFields.length; i++) {
+      const field = requiredFields[i];
+      if (!(field in req.body)) {
+        const message = `Missing \`${field}\` in request body`;
+        console.error(message);
+        return res.status(400).send(message);
+    }
+  }
+
+    let searched = req.body.title;
+    const url = `https://api.themoviedb.org/3/search/tv?api_key=40c781f4f82334b037fc6d9c33cc1c58&query=${searched}`;
+
+
+    fetch(url).then(response => {
+
+        if (!response.ok) {
+            Promise.reject(response.statusText);
```

Comment:
Oop, need a `return` here or it's not going to work. `Promise.reject` just returns a rejected promise, it doesn't abort your current promise chain or something.